aries_cloudagent/messaging/credential_definitions/routes.py

```diff
@@ -129,92 +143,160 @@ async def credential_definitions_send_credential_definition(request: web.BaseReq
+                            f"failed to upload: {reason}"
+                        )
+                    )
 
-        except RevocationNotSupportedError as e:
-            raise web.HTTPBadRequest(reason=e.message) from e
-        await shield(registry_record.generate_registry(session))
-        try:
-            await registry_record.set_tails_file_public_uri(
-                session, f"{tails_base_url}/{registry_record.revoc_reg_id}"
-            )
-            await registry_record.send_def(session)
-            await registry_record.send_entry(session)
+            except RevocationError as e:
+                raise web.HTTPBadRequest(reason=e.message) from e
 
-            # stage pending registry independent of whether tails server is OK
-            pending_registry_record = await revoc.init_issuer_registry(
-                registry_record.cred_def_id,
-                max_cred_num=registry_record.max_cred_num,
-            )
-            ensure_future(
-                pending_registry_record.stage_pending_registry(session, max_attempts=16)
-            )
+        return web.json_response({"credential_definition_id": cred_def_id})
 
-            tails_server = session.inject(BaseTailsServer)
-            (upload_success, reason) = await tails_server.upload_tails_file(
-                session,
-                registry_record.revoc_reg_id,
-                registry_record.tails_local_path,
-                interval=0.8,
-                backoff=-0.5,
-                max_attempts=5,  # heuristic: respect HTTP timeout
+    else:
+        transaction_message = await request.json()
+        session = await context.session()
+
+        wallet: BaseWallet = session.inject(BaseWallet, required=False)
+        if not wallet:
+            raise web.HTTPForbidden(reason="No wallet available")
+        author_did_info = await wallet.get_public_did()
+        if not author_did_info:
+            raise web.HTTPForbidden(
+                reason="Transaction cannot be created as there is no Public DID in wallet"
             )
-            if not upload_success:
-                raise web.HTTPInternalServerError(
-                    reason=(
-                        f"Tails file for rev reg {registry_record.revoc_reg_id} "
-                        f"failed to upload: {reason}"
-                    )
-                )
+        author_did = author_did_info.did
+        author_verkey = author_did_info.verkey
 
-        except RevocationError as e:
-            raise web.HTTPBadRequest(reason=e.message) from e
+        ledger: BaseLedger = session.inject(BaseLedger, required=False)
 
-    return web.json_response({"credential_definition_id": cred_def_id})
+        if not ledger:
+            reason = "No indy ledger available"
+            if not session.settings.get_value("wallet.type"):
+                reason += ": missing wallet-type?"
+            raise web.HTTPForbidden(reason=reason)
+
+        async with ledger:
```

Comment:
This code is all duplicated from existing code in the ledger object, is there some way to reuse (e.g. call ledger to create the transaction object) rather than repeating the code?